README.md

```diff
@@ -20,8 +20,8 @@ To make all the changes transparent and reproducible, `kustomize` is used to app
 
 Following notable commands/scripts are triggered in `make generate`:
 
-1. [`hack/generate-kustomize-patches.sh`](hack/fetch-upstream-released-manifest.sh)</br>
-    Fetch the upstream released manifest with the version specified in `helm/cluster-api/values.yaml`
+1. [`hack/fetch-manifest.sh`](hack/fetch-manifest.sh)</br>
+    Fetch the release manifest with the version specified in `helm/cluster-api/values.yaml`. Since we use our own fork of CAPI which does not replicate upstream GitHub releases, the YAML manifest is part of the Docker image as means to easily fetch the manifest which exactly matches the desired commit/tag.
```

Comment:
I believe that in our fork we also get GitHub releases with all manifests attached thanks to the GitHub action from the repo (at least that was the case before), it's just that the release is created as a draft and it should be published manually (it can be changed [here](https://github.com/giantswarm/cluster-api/blob/main/.github/workflows/release.yml#L35)) so those release artifacts can be fetched by our scripts.

IMO downloading a file from GitHub release (source of truth for manifests I believe) is simpler and easier than extracting manifests from Docker image.

Reply:
We would have to ask team shield about the security concerns.

Reply:
@giantswarm/team-shield can we get an opinion here?

Cluster API is using these actions [here](https://github.com/giantswarm/cluster-api/blob/main/.github/workflows/release.yml) (+ some basic bash and calling `make`):
- https://github.com/marketplace/actions/setup-go-environment
- https://github.com/marketplace/actions/checkout
- https://github.com/marketplace/actions/gh-release

> However using the actions means trusting external actions to run in our context.
>
> I think either works fine if we don't have security concerns.

IMO it should be fine to run these GitHub actions, as upstream Cluster API trusts them and runs these in upstream repo.

Reply:
Alternatively, we could run `make release-manifests` to generate manifests and use those.

Downside here, which applies also to Docker image approach, is that it's more likely that something could change in how this manifest generation works in upstream, so it could stop working for our use case. Nothing that cannot be fixed, but we would just have to be aware that we rely on some upstream dev process and track if it changes.

With GitHub release approach we simply rely on a manifest being attached in GitHub release, which is relying on a final output of the build/dev process, which is IMO more stable and has less maintenance.

Reply:
> I'm fine changing to GitHub releases to make things easier

@AndiDog that should mean not changing how we get manifests, so I think less work and changes here in the PR :smile: 

> GitHub actions are disabled at the moment

Although we would have to enable again these (not sure when and why these got disabled).

Reply:
It could have been me disabling the GitHub actions when switching to putting the manifests into images. I don't remember exactly.

GitHub releases are working, and the procedure is described in `cluster-api`'s README. It's not much more hassle, and consistent with upstream, so I'm good with this approach.

Reply:
> @team-shield can we get an opinion here?

Late to the party (time zones are hard) but I agree with your assessment.
We have only some basic guidelines for actions usage currently (but [want more](https://github.com/giantswarm/giantswarm/issues/12841)). There's some risk, but I'm personally ok with this one.